include/deal.II/base/index_set.h

```diff
@@ -229,6 +229,12 @@ class IndexSet
   void compress () const;
 
   /**
+   * Reorder the ranges inside IndexSet so that the largest range is the first
+   * one.
+   */
+  void reorder_ranges();
```

Comment:
Ah, reading through the comments for the entire PR, I see that I just re-invented the wheel, assuming what I wrote above is really what is implemented :-)